src/Microsoft.IdentityModel.Tokens/Validation/Results/Details/ExceptionDetail.cs

```diff
@@ -53,19 +57,60 @@ public ExceptionDetail(
         /// Creates an instance of an <see cref="Exception"/> using <see cref="ExceptionDetail"/>
         /// </summary>
         /// <returns>An instantance of an Exception.</returns>
-        public Exception GetException()
+        public virtual Exception GetException()
```

Comment:
The Activator used in this PR: https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/2800, is more extensible than this approach. Why are we resorting to still using a bunch of ifs checking for the different exceptions?

One more thing to consider is that the `ArgumentNullException` not inheriting from `SecurityTokenException` and therefore doesn't have a stack trace.